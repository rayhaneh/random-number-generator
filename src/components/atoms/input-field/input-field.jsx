import * as React from 'react';

export const InputField = (props) => {
  const {
    handleChange,
    value,
    min,
    max,
  } = props;

  return (
    <input
      className={'pa2 ba b--blue blue'}
      onChange={handleChange}
      type='number'
      value={value}
      min={min}
      max={max}
    />
  );
}
