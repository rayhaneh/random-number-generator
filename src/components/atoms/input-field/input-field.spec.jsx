import * as React from 'react';
import {
  configure,
  shallow,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
configure({ adapter: new Adapter() });

import { InputField } from './input-field';

let props;
let comp;
let mockHandleChange = jest.fn();

describe('InputField Component', () => {
  beforeEach(() => {
    props = {
      placeholder: 'min',
      value: '',
      handleChange: mockHandleChange,
    }
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders properly', () => {
    comp = shallow(<InputField {...props} />);
    expect(comp).toMatchSnapshot();
  });

  it('calls handleChange when input is changed', () => {
    comp = shallow(<InputField {...props} />);
    comp.find('input')
      .first()
      .simulate('change', 3);
    expect(mockHandleChange.mock.calls.length).toBe(1);
  });
});
