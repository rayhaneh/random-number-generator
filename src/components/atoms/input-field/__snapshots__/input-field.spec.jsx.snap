// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField Component renders properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <InputField
    handleChange={[MockFunction]}
    placeholder="min"
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "pa2 ba b--blue blue",
      "max": undefined,
      "min": undefined,
      "onChange": [MockFunction],
      "type": "number",
      "value": "",
    },
    "ref": null,
    "rendered": null,
    "type": "input",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "pa2 ba b--blue blue",
        "max": undefined,
        "min": undefined,
        "onChange": [MockFunction],
        "type": "number",
        "value": "",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
